libPhoneNumber/NBPhoneNumber.m

```diff
@@ -48,8 +48,12 @@ - (BOOL)isEqualToObject:(NBPhoneNumber*)otherObj
 
 - (NSUInteger)hash
 {
-    NSData *selfObject = [NSKeyedArchiver archivedDataWithRootObject:self];
-    return [selfObject hash];
+    NSUInteger hash = 0;
+    hash ^= self.countryCode.hash + 0x9e3779b9 + (hash << 6) + (hash >> 2);
```

Comment:
Given that hash is going to be 0 at this point, there really isn't any point in the shift left 6 and shift right 2 here is there? Can you remove that? 

Also could you add a link to https://stackoverflow.com/questions/4948780/magic-number-in-boosthash-combine in a comment to describe where you got the hash from and where the magic number came from?